llvm/lib/Target/DirectX/DXILShaderFlags.cpp

```diff
@@ -36,8 +39,35 @@ static void updateFlags(ComputedShaderFlags &Flags, const Instruction &I) {
   }
 }
 
-ComputedShaderFlags ComputedShaderFlags::computeFlags(Module &M) {
+static void updateResourceFlags(ComputedShaderFlags &Flags, Module &M,
+                                ModuleAnalysisManager *AM) {
+  if (!AM)
+    return;
+
+  const DXILResourceMap &DRM = AM->getResult<DXILResourceAnalysis>(M);
+  if (DRM.empty())
+    return;
+
+  const dxil::ModuleMetadataInfo &MMDI = AM->getResult<DXILMetadataAnalysis>(M);
+  VersionTuple SM = MMDI.ShaderModelVersion;
+  Triple::EnvironmentType SP = MMDI.ShaderProfile;
+
+  // RWBuffer
+  for (const ResourceInfo &RI : DRM.uavs()) {
+    if (RI.getResourceKind() == ResourceKind::TypedBuffer) {
```

Comment:
This looks wrong.  It should only set this if it's a `StructuredBuffer` or `ByteAddressBuffer` (aka. raw buffer), but not if it's a `TypedBuffer`.

Reply:
Hmm.. I went for this approach because of how the IR looked.  Maybe I missed something here.
```
; RWBuffer<int> Buf : register(u7, space2)
  %uav0 = call target("dx.TypedBuffer", i32, 1, 0, 1)
      @llvm.dx.handle.fromBinding.tdx.TypedBuffer_i32_1_0t(
          i32 2, i32 7, i32 1, i32 0, i1 false)
```

Reply:
In HLSL: `RWBuffer` is a typed buffer.  You want to set this if one of the HLSL types: `RWStructuredBuffer` or `RWByteAddressBuffer` are used.

Reply:
I believe I fixed this now.